packages/deployments/contracts/contracts/core/connext/libraries/AssetLogic.sol

```diff
@@ -22,12 +22,26 @@ library AssetLogic {
   error AssetLogic__handleIncomingAsset_nativeAssetNotSupported();
   error AssetLogic__handleIncomingAsset_feeOnTransferNotSupported();
   error AssetLogic__handleOutgoingAsset_notNative();
-  error AssetLogic__swapToLocalAssetIfNeeded_swapPaused();
-  error AssetLogic__swapFromLocalAssetIfNeeded_swapPaused();
   error AssetLogic__getTokenIndexFromStableSwapPool_notExist();
+  error AssetLogic__getConfig_notRegistered();
 
   // ============ Internal: Handle Transfer ============
 
+  function getConfig(bytes32 _key) internal view returns (TokenConfig storage) {
+    AppStorage storage s = LibConnextStorage.connextStorage();
+    TokenConfig storage config = s.tokenConfigs[_key];
+
+    // Sanity check: not empty
+    // NOTE: adopted decimals will *always* be nonzero (or reflect what is onchain
+    // for the asset). The same is not true for the representation assets, which
+    // will always have 0 decimals on the canonical domain
+    if (config.adoptedDecimals < 1) {
```

Comment:
would it be better to just check `config.approval` here?

Reply:
you may want the config for a non-approved asset